lib/matobjplist.gi

```diff
@@ -329,10 +308,7 @@ InstallMethod( Unpack, "for a plist vector",
 
 InstallMethod( ShallowCopy, "for a plist vector", [ IsPlistVectorRep ],
   function( v )
-    local res;
-    res := Objectify(TypeObj(v),[v![BDPOS],ShallowCopy(v![ELSPOS])]);
-    if not IsMutable(v) then SetFilterObj(res,IsMutable); fi;
-    return res;
+    return MakeIsPlistVectorRep(v![BDPOS], true, ShallowCopy(v![ELSPOS]));
```

Comment:
Doesn't `ShallowCopy` return `v` when `v` is immutable, and only return a copy when `v` is mutable?

Reply:
If `v` is in the filter `IsCopyable` then `ShallowCopy( v )` returns always a new object which is mutable.
If `v` is not in the filter `IsCopyable` then `ShallowCopy( v )` returns `v`.

Reply:
Thanks @ThomasBreuer I didn't know this!

Reply:
see https://docs.gap-system.org/doc/ref/chap12_mj.html#X846BC7107C352031

(`IsCopyable` seems to be a not well-known concept in GAP. Perhaps the documentation should better explicitly mention `IsCopyable` instead of talking about "copyable objects" or "objects that support a mutable form".)

Reply:
Yeah, I've been working on GAP for years but only realized in 2022 (when Thomas told me) what it really does; I somehow never had read its documentation entry. It really means `IsAbleToMakeMutableInstances` or `CanHaveMutableInstances` so (I am not proposing a renaming, just sayin')

Reply:
I've removed the `mutable` argument for `MakeIsPlistVectorRep` which should take care of this and other issues.